manifests/repo.pp

```diff
@@ -5,126 +5,160 @@
+#   Set the name parameter of the repository configuration. Defaults to the name
+#   of the defined type.
+#
+# @param environment
+#   Specify the environment part of the default value for baseurl (see $baseurl).
+#   Specifying $baseurl or $mirrorlist will override this parameter.
+#
+# @param mirrorlist
+#   Trigger to set the mirrorlist parameter of the repository configuration.
+#
+# @param failovermethod
+#   Trigger to set the failovermethod parameter of the repository configuration.
+#
+# @param sslcacert
+#   If set, will ensure the line sslcacert is present in the repository
+#   configuration with the specified value. This is useful when using your own
+#   CA bundle.
 #
 define yum::repo (
-  $ensure               = 'present',
-  $baseurl              = undef,
-  $enabled              = true,
-  $gpgcheck             = true,
-  $gpgkey               = undef,
-  $use_gpgkey_uri       = true,
-  $priority             = undef,
-  $repo_server          = "yum.${::domain}",
-  $repo_server_protocol = 'http',
-  $repo_server_basedir  = '/',
-  $repo_file_mode       = '0400',
-  $yum_repos_d_path     = '/etc/yum.repos.d',
-  $gpgkey_url_proto     = 'http',
-  $gpgkey_url_server    = undef,
-  $gpgkey_url_path      = 'keys',
-  $gpgkey_file_prefix   = 'RPM-GPG-KEY',
-  $gpgkey_local_path    = '/etc/pki/rpm-gpg',
-  $username             = undef,
-  $password             = undef,
-  $description          = undef,
-  $environment          = "${::environment}", # lint:ignore:only_variable_string # stringification needed for spec testing on Puppet 3.8 & 4.3.0
-  $mirrorlist           = undef,
-  $failovermethod       = undef,
-  $sslcacert            = undef,
+  Enum['absent', 'present'] $ensure         = 'present',
+  Optional[String] $baseurl                 = undef,
+  Boolean $enabled                          = true,
+  Boolean $gpgcheck                         = true,
+  Optional[String] $gpgkey                  = undef,
+  Boolean $use_gpgkey_uri                   = true,
+  Optional[Integer] $priority               = undef,
+  String $repo_server                       = "yum.${::domain}",
+  String $repo_server_protocol              = 'http',
+  String $repo_server_basedir               = '/',
+  Pattern[/^[0-7]{4}$/] $repo_file_mode     = '0400',
+  Stdlib::Absolutepath $yum_repos_d_path    = '/etc/yum.repos.d',
+  String $gpgkey_url_proto                  = 'http',
+  String $gpgkey_url_server                 = "yum.${::domain}",
+  String $gpgkey_url_path                   = 'keys',
+  String $gpgkey_file_prefix                = 'RPM-GPG-KEY',
+  Stdlib::Absolutepath $gpgkey_local_path   = '/etc/pki/rpm-gpg',
+  Optional[String] $username                = undef,
+  Optional[String] $password                = undef,
+  Optional[String] $description             = $name,
+  String $environment                       = "${::environment}", # lint:ignore:only_variable_string # stringification needed for spec testing on Puppet 3.8 & 4.3.0
```

Comment:
Do we still need the stringification noted in the comment?

Reply:
nope, removed